src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -531,6 +542,10 @@ def __init__(
         # on the optimized plan. During the final query generation, no schema query is needed,
         # this helps reduces un-necessary overhead for the describing call.
         self._skip_schema_query = skip_schema_query
+        self._config_context: Optional[ConfigContext] = None
```

Comment:
The config context here should probably been done in the same way as what we have done for analyze and resolve as parameter for each builder.